### DISCUSSION TITLE
Adding run failure sensors to software-defined asset jobs

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#7774] [Q&A]
The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1651758787895909?thread_ts=1651758787.895909&cid=C01U954MEER). Reposting it here for posterity and discoverability.

### Original question:

Is there a way to attach a hook to a job built from a software-defined asset?

**Comment by prha** [comment]
If you are looking for a way to fire an alert or perform some action based on a run failure of an asset job, you can set up a [run status sensor](https://docs.dagster.io/concepts/partitions-schedules-sensors/sensors#run-status-sensors).  You can point it at a specific job, including an asset job.  For example:

```
from dagster import asset, build_assets_job, run_failure_sensor

@asset
def asset1():
    return 1

@asset
def asset2():
    return 2

my_asset_job = build_assets_job("my_assets_job", assets=[asset1, asset2])

@run_failure_sensor(job_selection=[my_asset_job])
def perform_on_asset_job_failure(context):
    ...
```
